File: go/darts/darts.go
package darts

import "math"

func Score(x float64, y float64) int {

	dist := math.Pow(math.Pow(x, 2)+math.Pow(y, 2), 0.5)

	if dist <= 1 {
		return 10
	}

	if dist <= 5 {
		return 5
	}

	if dist <= 10 {
		return 1
	}
	return 0

}

File: ruby/darts/darts.rb
class Darts
  def initialize (x, y)
    @x = x
    @y = y
    @dist = ( @x * @x + @y * @y ) ** 0.5
  end

  def score
    @dist > 10 ? 0 : @dist > 5 ? 1 : @dist > 1 ? 5 : 10
  end
end
